src/config/load.c

```diff
@@ -357,12 +357,26 @@ cfgLoadUpdateOption(void)
     if (cfgOptionValid(cfgOptCompressType))
         compressTypePresent(compressTypeEnum(cfgOptionStrId(cfgOptCompressType)));
 
-    // Update compress-level default based on the compression type
-    if (cfgOptionValid(cfgOptCompressLevel) && cfgOptionSource(cfgOptCompressLevel) == cfgSourceDefault)
+    // Update compress-level default based on the compression type. Also check that level range is valid per compression type.
+    if (cfgOptionValid(cfgOptCompressLevel))
     {
-        cfgOptionSet(
-            cfgOptCompressLevel, cfgSourceDefault,
-            VARINT64(compressLevelDefault(compressTypeEnum(cfgOptionStrId(cfgOptCompressType)))));
+        const CompressType compressType = compressTypeEnum(cfgOptionStrId(cfgOptCompressType));
+
+        if (cfgOptionSource(cfgOptCompressLevel) == cfgSourceDefault)
+        {
+            cfgOptionSet(cfgOptCompressLevel, cfgSourceDefault, VARINT64(compressLevelDefault(compressType)));
+        }
+        else if (compressType != compressTypeNone)
+        {
+            if (cfgOptionInt(cfgOptCompressLevel) < compressLevelMin(compressType) ||
+                cfgOptionInt(cfgOptCompressLevel) > compressLevelMax(compressType))
+            {
+                THROW_FMT(
+                    OptionInvalidValueError,
+                    "'%d' is out of range for '" CFGOPT_COMPRESS_LEVEL "' option when '" CFGOPT_COMPRESS_TYPE " option ' = '%s'",
```

Comment:
This gives `ERROR: [032]: '21' is out of range for 'compress-level' option when 'compress-type option ' = 'gz'`

Shouldn't it be `when '" CFGOPT_COMPRESS_TYPE "' option = '%s'`?

Reply:
Hmm, even after a77a016, I still think there's an `'` too much:

Shouldn't `when '" CFGOPT_COMPRESS_TYPE "' option ' = '%s'` become `when '" CFGOPT_COMPRESS_TYPE "' option = '%s'` ?

I can try to change it myself if you wish ;-)

Reply:
Wow, tough day! Fixed (I think) in aada0a951e367dbf.

Reply:
Indeed, looks good to me now :-) :+1: